test/test_dir.rs

```diff
@@ -0,0 +1,46 @@
+extern crate nix;
+extern crate tempdir;
+
+use nix::dir::{Dir, Type};
+use nix::fcntl::OFlag;
+use nix::sys::stat::Mode;
+use std::fs::File;
+use self::tempdir::TempDir;
+
+#[test]
+fn read() {
+    let tmp = TempDir::new("nix-dir-read").unwrap();
+    File::create(&tmp.path().join("foo")).unwrap();
+    ::std::os::unix::fs::symlink("foo", tmp.path().join("bar")).unwrap();
+    let mut dir = Dir::open(tmp.path(), OFlag::O_DIRECTORY | OFlag::O_RDONLY | OFlag::O_CLOEXEC,
+                            Mode::empty()).unwrap();
+    let mut entries: Vec<_> = dir.iter().map(|e| e.unwrap()).collect();
+    entries.sort_by(|a, b| a.file_name().cmp(b.file_name()));
+    let entry_names: Vec<_> = entries
+        .iter()
+        .map(|e| e.file_name().to_str().unwrap().to_owned())
+        .collect();
+    assert_eq!(&entry_names[..], &[".", "..", "bar", "foo"]);
+
+    // Check file types. The system is allowed to return DT_UNKNOWN (aka None here) but if it does
+    // return a type, ensure it's correct.
+    assert!(&[Some(Type::Directory), None].contains(&entries[0].file_type())); // .: dir
+    assert!(&[Some(Type::Directory), None].contains(&entries[1].file_type())); // ..: dir
+    assert!(&[Some(Type::Symlink), None].contains(&entries[2].file_type())); // bar: symlink
+    assert!(&[Some(Type::File), None].contains(&entries[3].file_type())); // foo: regular file
+}
+
+#[test]
+fn rewind() {
+    let tmp = TempDir::new("nix-dir-rewind").unwrap();
```

Comment:
Can't we have anonymous folders? I'd prefer we use that style rather than naming everything.

Reply:
The `tempdir` crate doesn't seem to support that. `tempfile` version 3 [does](https://docs.rs/tempfile/3.0.3/tempfile/fn.tempdir.html) (and that obsoletes `tempdir`), but nix still seems to be on version 2. This call to `TempDir::new` matches the other ones in the `tests` dir.